sdk/core/core/src/az_json_data.c

```diff
@@ -0,0 +1,220 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#include <az_json_data.h>
+
+#include <az_json_parser.h>
+#include <az_span_builder.h>
+
+#include <_az_cfg.h>
+
+typedef struct {
+  az_span span;
+  size_t align;
+} az_data;
+
+#define AZ_DATA(V) \
+  ((az_data){ \
+      .span = (az_span){ .begin = (uint8_t const *)&(V), .size = sizeof(V) }, \
+      .align = AZ_ALIGNOF(V), \
+  })
+
+typedef struct {
+  size_t size;
+  size_t align;
+} _az_type;
+
+#define _AZ_TYPE(T) \
+  ((_az_type){ \
+      .size = sizeof(T), \
+      .align = AZ_ALIGNOF(T), \
+  })
+
+uint8_t * _az_align_floor(uint8_t * const p, size_t const align) {
+  return (uint8_t *)(((size_t)p) / align * align);
+}
+
+uint8_t * _az_align_ceil(uint8_t * const p, size_t const align) {
+  return _az_align_floor(p + (align - 1), align);
+}
+
+AZ_NODISCARD az_result _az_span_builder_aligned_append(
```

Comment:
I'd like to see comments describing how these functions align data (where the padding goes)